### PR TITLE
Feat/add inventory discovery

### PR DESCRIPTION
**Pull request by abarciauskas-bgse**
Opening this before self-review so work is not lost, but will self-review before opening "ready" PR.

### CODE REVIEW

**Comment by emileten** [comment]
@abarciauskas-bgse thanks ! Question, are we not using the inventory file at the moment in `main`?

**Comment by abarciauskas-bgse** [comment]
@emileten good question - I have deployed these changes but they haven't been merged yet 😬

File: OPERATING.md [hunk 1]
```diff
@@ -82,6 +82,11 @@ Should follow the following format:
     "temporal": ["<start-date>", "<end-date>"],
     "bounding_box": ["<bounding-box-as-comma-separated-LBRT>"],
     "include": "<filename-pattern>",
+
+    ## for inventory
+    "inventory_url": "s3://...",
+    "filename_regex": "<filename-regex>",
```
Comment (emileten) [inline OPERATING.md:88]: The way this is written, it makes me believe the `filename_regex` key is a required parameter to make use of the inventory feature. However, it does not appear in the example step function input json file for icesat2-boreal -- you actually removed it, it had the value "^(.*).tif$". Is this now a default value somewhere for that parameter ?

File: lambdas/data-transfer/handler.py [hunk 2]
```diff
@@ -43,6 +43,10 @@ def handler(event, context):
         filename = src_key.split("/")[-1]
 
         target_key = f"{object.get('collection')}/{filename}"
+        # The file-staging directory is the default for MAAP's bucket
```
Comment (emileten) [inline lambdas/data-transfer/handler.py:46]: How was this working before, in order to have the file-staging folder actually show up in the `target_url`? 

My guess is it was contained in the `TARGET_BUCKET` string ?

File: lambdas/inventory/handler.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
+import json
```
Comment (emileten) [inline lambdas/inventory/handler.py:1]: Tell me if I am wrong, but quickly looking at this I feel this makes some assumptions about how the inventory data is stored. Are we documenting these constraints somewhere?

**Comment by emileten** [comment]
@abarciauskas-bgse I am merging this because I need the inventory feature for https://github.com/NASA-IMPACT/active-maap-sprint/issues/461. If we find issues we can re-iterate on this.

**Pull request merged**